errors.go

```diff
@@ -296,6 +317,7 @@ func Augment(err error, context string, params map[string]string) error {
 		return &Error{
 			Code:         err.Code,
 			Message:      context,
+			MessageChain: append([]string{err.Message}, err.MessageChain...),
```

Comment:
How about doing also populating this in other "wrapping" function like Wrap and NewInternalWithCause?

Reply:
I don't have a super strong opinion but my reasoning against this is:
* Wrap is deprecated and I'd prefer to make no changes to deprecated functionality
* NewInternalWithCause is an explicit instruction to create a new error.  Augment is more an instruction to pass an error along but tack on a bit of additional info.  The problem with Augment though is that it wasn't really doing this, it was (in a way) creating a new error.

Reply:
Fair enough regarding NewInternalWithCause. I'd probably rather do it in Wrap too though just because it's still used 11264 in the wearedev codebase (vs 55747 for Augment) and the different behaviour might lead to confusion. I can see arguments both ways though - so up to you.

Reply:
I just looked at this for `Wrap` and I'm not clear on what to add to the `MessageChain`.  This function isn't like `Augment` in that it associates two errors, it just returns a new `Error` which may have more params than the original (the `Message` for the new error will be the same as the old one).